test/index.test.ts

```diff
@@ -863,6 +863,21 @@ describe('Unit | Utility | changeset', () => {
         initialData = { contact: { emails: ['bob@email.com'] } };
       });
 
+      it('works with boolean values', () => {
+        let initialData = { contact: { emails: [{}, {}] } };
+
+        const changeset = Changeset(initialData);
+
+        changeset.set('contact.emails.2', { nested: false });
+        expect(changeset.get('contact.emails.2')).toEqual({ nested: false });
```

Comment:
before this PR, this assertion failed because the value that represented `nested` was `undefined`